packages/apis/src/lib.rs

```diff
@@ -1,4 +1,30 @@
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{Timestamp, Uint64};
+
 pub mod id;
 pub mod interfaces;
 pub mod tasks;
 pub mod verifier_simple;
+
+/// A wrapper around u64, that represents duration between two points in time.
+#[cw_serde]
+#[derive(Copy, PartialOrd, Ord, Eq)]
+pub struct Nanos(Uint64);
```

Comment:
Please rename it, the comment alone tells me way more about the purpose of it rather than the name of the structure alone.
```suggestion
/// A wrapper around u64, that represents duration between two points in time.
#[cw_serde]
#[derive(Copy, PartialOrd, Ord, Eq)]
pub struct Duration(Uint64);
```
You can do an alias if for better clarity as well.
```rust
type Nanos = u64;
```

Reply:
Please check this duration implementation: https://github.com/confio/poe-contracts/blob/b7a8dbafd89cd70401dced518366f520b7089ff6/packages/utils/src/time.rs#L6-L9

Reply:
added it, although I think we can trim it out a bit - i.e. remove `Expiration` and the fns associated with it

Reply:
The problem with `Duration` is that `cosmwasm_std::Timestamp` doesn't have an `add_duration`, and so we end up back with the original problem of not knowing what unit of duration we're talking about.

Changing to `Duration` makes sense, but please change `.u64()` to `.nanos()` and `new()` to `new_nanos()` so that we can see from the code itself what unit we're working with and remove the footgun of accidentally using seconds when we mean nanoseconds.